tests/savefiles.spec.ts

```diff
@@ -1,66 +1,98 @@
+  if (!await editAttributesLocator.isDisabled()) {
+    await editAttributesLocator.click();
+  }
+
+  const clearAllLocator = page.getByRole("button", { name: "Clear all" });
+  if (!await clearAllLocator.isDisabled()) {
+    await clearAllLocator.click();
+  }
+
+  if (!await editAttributesLocator.isDisabled()) {
+    await editAttributesLocator.click();
+  }
+});
+
+test("loading a file with length displays the length", async () => {
   // Note we use a weird styling trick to replace the file input element with a regular button.
   // This will not work: page.getByRole('button', { name: 'Load from GeoJSON' })
-  await page.locator("#load_geojson").setInputFiles("tests/data/route.json");
+  await page.locator("#load_geojson").setInputFiles("tests/data/Adur.json");
 
-  await page.getByRole("button", { name: "1) Untitled route" }).click();
-  await expect(page.getByText("Length: 8.5km")).toBeVisible();
+  page.on('dialog', (dialog) => dialog.accept());
+  await page.getByRole("button", { name: "1) Route from Dankton Lane and West Street to Cokeham Road and Western Road North" }).click();
+  await expect(page.getByText("Length: 450 m")).toBeVisible();
 });
 
 // Some people used ATIP before route-snapper started populating a length
 // property. Upload a file with that missing, and make sure it's backfilled.
-test("loading a file without length displays the length", async ({ page }) => {
+test("loading a file without length displays the length", async () => {
   // Remove the property from the test data first
-  let json = JSON.parse(
-    await readFile("tests/data/route.json", { encoding: "utf8" })
-  );
   await expect(Math.round(json.features[0].properties.length_meters)).toBe(
-    8520
+    450
   );
+  const lengthMeters = json.features[0].properties.length_meters;
   delete json.features[0].properties.length_meters;
   let uploadFile = JSON.stringify(json);
 
-  await page.goto("/scheme.html?authority=North Somerset");
   await page.locator("#load_geojson").setInputFiles({
-    name: "route.json",
+    name: "Adur.json",
     mimeType: "application/json",
     buffer: Buffer.from(uploadFile),
   });
 
-  await page.getByRole("button", { name: "1) Untitled route" }).click();
-  await expect(page.getByText("Length: 8.5km")).toBeVisible();
+  json.features[0].properties.length_meters = lengthMeters;
+
+  await page.getByRole("button", { name: "1) Route from Dankton Lane and West Street to Cokeham Road and Western Road North" }).click();
+  await expect(page.getByText("Length: 450 m")).toBeVisible();
 });
 
 // Handle unusual GeoJSON inputs produced by other tools
 test("loading a file with null properties displays the length", async ({
   page,
 }) => {
   // Remove the property from the test data first
-  let json = JSON.parse(
-    await readFile("tests/data/route.json", { encoding: "utf8" })
-  );
+  const properties = json.features[0].properties;
   json.features[0].properties = null;
```

Comment:
This mutates the shared `json`. I think we have to be paranoid and make a deep copy in here. `JSON.parse(JSON.stringify(json))` is the hack I've been using everywhere, but supposedly https://web.dev/structured-clone/ is the new way

Reply:
ahh a classic